File: Problems/Algorithms/1345. Check If N and Its Double Exist/CheckDouble.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer> set = new HashSet<>();
        
        for (int i = 0; i < arr.length; i++) {
            int curr = arr[i];
            if (set.contains(2*curr)) return true;
            if (curr % 2 == 0 && set.contains(curr/2)) return true;
            set.add(arr[i]);
        }
        
        return false;
    }
}

File: Explore/Learn/Introduction to Data Structure/Arrays 101/Check If N and Its Double Exist/checkDouble.kt
class Solution {
    fun checkIfExist(arr: IntArray): Boolean {
        val length = arr.size
        if (length <= 1) return false
        
        for (i in 0..(length-1)) {
            for (j in (i+1)..(length-1)) {
                if (arr[j] == 2 * arr[i]) return true
            }
            if (arr[i] % 2 == 0) {
                for (j in (i+1)..(length-1)) {
                    if (arr[j] == arr[i] / 2) return true
                }
            }
        }
        
        return false
    }
}

